bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/PersistentEntryLogMetadataMap.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorage.CloseableIterator;
+import org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorageFactory.DbConfigType;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Persistent entryLogMetadata-map that stores entry-loggers metadata into
+ * rocksDB.
+ */
+public class PersistentEntryLogMetadataMap implements EntryLogMetadataMap {
+    private static final Logger LOG = LoggerFactory.getLogger(PersistentEntryLogMetadataMap.class);
+    // persistent Rocksdb to store metadata-map
+    private final KeyValueStorage metadataMapDB;
+    private AtomicBoolean isClosed = new AtomicBoolean(false);
+
+    private static final FastThreadLocal<ByteArrayOutputStream> baos = new FastThreadLocal<ByteArrayOutputStream>() {
+        @Override
+        protected ByteArrayOutputStream initialValue() {
+            return new ByteArrayOutputStream();
+        }
+    };
+    private static final FastThreadLocal<ByteArrayInputStream> bais = new FastThreadLocal<ByteArrayInputStream>() {
+        @Override
+        protected ByteArrayInputStream initialValue() {
+            return new ByteArrayInputStream(new byte[1]);
+        }
+    };
+    private static final FastThreadLocal<DataOutputStream> dataos = new FastThreadLocal<DataOutputStream>() {
+        @Override
+        protected DataOutputStream initialValue() {
+            return new DataOutputStream(baos.get());
+        }
+    };
+    private static final FastThreadLocal<DataInputStream> datais = new FastThreadLocal<DataInputStream>() {
+        @Override
+        protected DataInputStream initialValue() {
+            return new DataInputStream(bais.get());
+        }
+    };
+
+    public PersistentEntryLogMetadataMap(String path, ServerConfiguration conf) throws IOException {
+        String metadataPath = FileSystems.getDefault().getPath(path, "metadata").toFile().toString();
+        metadataMapDB = KeyValueStorageRocksDB.factory.newKeyValueStorage(metadataPath, DbConfigType.Small, conf);
+    }
+
+    @Override
+    public boolean containsKey(long entryLogId) throws EntryLogMetadataMapException {
+        LongWrapper key = LongWrapper.get(entryLogId);
+        try {
+            boolean isExist;
+            try {
+                isExist = metadataMapDB.get(key.array) != null;
+            } catch (IOException e) {
+                throw new EntryLogMetadataMapException(e);
+            }
+            return isExist;
+        } finally {
+            key.recycle();
+        }
+    }
+
+    @Override
+    public void put(long entryLogId, EntryLogMetadata entryLogMeta) throws EntryLogMetadataMapException {
+        LongWrapper key = LongWrapper.get(entryLogId);
+        try {
+            baos.get().reset();
+            try {
+                entryLogMeta.serialize(dataos.get());
+                dataos.get().flush();
```

Comment:
output of serialize is expected to be flushed. so move this flush call to serialize method.